article/background.tex

```diff
@@ -4,107 +4,69 @@ \section{Background}
+Therefore, the authors consider all code that is executed TODO:(not neccesarily OS,etc tone it down) to be part of the requisite analysis instructions.
+This includes code that is installed as dependencies, which may come from many repositories that cannot be relied upon to host all legacy versions.
+Reexecution either requires updating the software to modern dependencies, or the creation of a similar computational environment.
+Environment construction becomes harder to reproduce with time as upstream repositories update, as the experienced with this project.
+
+\subsection{Software Dependency Problem and Containers}
+
+A fundamental challenge of software dependency management is that dependencies can be nested arbitrarily deeply, which increases in complexity (TODO: O^n? does that even make sense to say?)
+As soon as one version of one dependency eventually becomes unavailable, the first step of executability is manually solving a computationally difficult problem.
+One way of solving the issue of rebuilding environments with integrated dependencies is distributing an entire operating system image with the environment installed and configured, which can be emulated as a "guest" environment within the "host" machine.
+Virtual machines (colloquially "VMs") are attractive solutions to the software dependency problem, because the artifact that is distributed is in working order and should be executable with almost no knowledge of the internal workings.
+Once running, guest machines are self-contained and isolated from the host, which then eliminates the posibility of polluting the host environment and allows system administrators to safely allow relatively unrestricted usage from semi-trusted users.
+The weakness of VMs for solving this problem is primarily that they are expensive; VMs are inefficient with resource allocation in general. \cite{TODO: asmacdo} \footnote{Hard drive usage(and bandwidth), memory allocation, CPU allocation, start/restart time}
+
+Modern advances in "container technology" have allowed similar benefits but strip redundancy by making limited use of the host machine, specifically the hypervisor.
+Containers enable a complete working environment often as small as a few Megabytes, and can it be started as quickly as a normal local process.
+Over time Docker and other organizations have established a governing body under the Linux Foundation project, the "Open Container Initiative" (OCI).
+OCI produced an specification for containers, which can be used by various container runtimes and toolsets including Docker, Podman, and Apptainer \cite{TODO: apptainer something}
+While OCI images are nearly ubiquitous in the cloud industry, Singularity (recently renamed to Apptainer) is a toolset that was developed for High Performance Computing, and has its own native specification.
+
+\subsection{Preserving Version Collection of Modular Components}
```

Comment:
I do not understand what this subsection title means. Is this terminology from YODA? “Provenance Tracking“ could be better maybe?

Reply:
Not YODA, Im just trying to describe what we are specifically doing. PRovenance tracking is too vague of a term IMO, since I am specifically discussing the versions of the modular components, and not discussing tracking each command to be run etc. WDYT about "Preserving a Collection of Versioned Components"